### PR TITLE
SUS-78 | Special:Videos remove sort dropdown

### PR DESCRIPTION
**Pull request by kvas-damian**
https://wikia-inc.atlassian.net/browse/SUS-78
- removed sort dropdown on [Special/Videos](http://community.wikia.com/wiki/Special:Videos)
- removed `viewCount` from `FilePageController` and `FilePageController`
- [API `SpecialVideosSpecialController` method `index`](http://elderscrolls.wikia.com/wikia.php?controller=SpecialVideosSpecial&method=index&sort=trend&format=json&provider=youtube,ooyala)
  - request - ignore sort parameter
  - response removed: `sortMsg` and `sortingOptions`
  - `viewsTotal` stays untouched because it's used by Mobile Apps, but it returns stale data
- [ API `VideoHandler` method `getVideoList`](http://elderscrolls.wikia.com/wikia.php?controller=VideoHandler&method=getVideoList&sort=popular&limit=4&format=json&providers[]=youtube&providers[]=ooyala&detail=1) - ignore sort parameter

cc: @Wikia/sustaining-team @aniaMu

### CODE REVIEW

File: extensions/wikia/SpecialVideos/SpecialVideosHelper.class.php [hunk 1]
```diff
@@ -46,69 +46,17 @@ public function getMetaTagDescription() {
...
 	 * get list of videos
-	 * @param string $sort [recent/popular/trend]
 	 * @param integer $page
+	 * @param string $filter [all/premium]
 	 * @param array $providers
 	 * @param string $category
 	 * @param array $options
 	 * @return array $videos
 	 */
-	public function getVideos( $sort, $page, $providers = [], $category = '', $options = [] ) {
```
Comment (ghost) [inline extensions/wikia/SpecialVideos/SpecialVideosHelper.class.php:102]: Did you check usage of this method. We are breaking backwards compatibility here.

File: extensions/wikia/SpecialVideos/SpecialVideosHelper.class.php [hunk 2]
```diff
@@ -46,69 +46,17 @@ public function getMetaTagDescription() {
...
+	public function getVideos( $page, $filter = 'all', $providers = [], $category = '', $options = [] ) {
 		wfProfileIn( __METHOD__ );
 
-		if ( $sort == 'premium' ) {
```
Comment (ghost) [inline extensions/wikia/SpecialVideos/SpecialVideosHelper.class.php:105]: OMG, that was so :hammer:

File: extensions/wikia/SpecialVideos/SpecialVideosHelper.class.php (hunk 1 above)
Comment (kvas-damian) [inline extensions/wikia/SpecialVideos/SpecialVideosHelper.class.php:102]: Yep, I did found only one usage: `SpecialVideosSpecialController::getVideos`

File: extensions/wikia/SpecialVideos/SpecialVideosHelper.class.php [hunk 3]
```diff
@@ -149,7 +97,7 @@ public function getVideos( $sort, $page, $providers = [], $category = '', $optio
 			$videoDetail = $helper->getVideoDetail( $videoInfo, $videoOptions );
 			if ( !empty( $videoDetail ) ) {
 				$byUserMsg = WikiaFileHelper::getByUserMsg( $videoDetail['userName'], $videoDetail['timestamp'] );
-				// FIXME: this isn't shown to users b/c it's buggy (CONSF-51)
+				// SUS-78 | Not used in template but used by API clients - GameGuides App
 				$viewTotal = wfMessage( 'videohandler-video-views', $this->wg->Lang->formatNum( $videoDetail['viewsTotal'] ) )->text();
```
Comment (ghost) [inline extensions/wikia/SpecialVideos/SpecialVideosHelper.class.php]: Isn't that connected to https://github.com/Wikia/app/pull/9673/files#diff-e10911b408f376068e7330dad13f1fd2L548?

Comment (kvas-damian) [inline extensions/wikia/SpecialVideos/SpecialVideosHelper.class.php]: It's controller code, it's not related. According to docs mobileApps aren't using it.

**Comment by ghost** [comment]
:+1: Let's QA this. Code-wise it is fine.

**Comment by ghost** [comment]
:+1:

**Pull request merged**